Build: Simplify how we invoke webpack

flootr opened the pull request:
• ~~Use exposed `webpack` bin from `node_modules/.bin`~~ don't, to ensure win compat
• Omit `--config webpack.config.js` because that's the default

flootr committed:
Use 'webpack' bin from node modules bin dir

blowery reviewed (dismissed):
LGTM! Nice simplification.

jsnajdr commented on package.json:
```diff
@@ -206,7 +206,7 @@
     "prebuild-server": "mkdirp build",
     "build-server": "cross-env-shell CALYPSO_SERVER=true NODE_PATH=$NODE_PATH:server:client:. webpack --display-error-details --config webpack.config.node.js",
-    "build-client": "cross-env-shell CALYPSO_CLIENT=true NODE_PATH=$NODE_PATH:server:client:. node --max_old_space_size=8192 ./node_modules/webpack/bin/webpack.js --display-error-details --config webpack.config.js",
+    "build-client": "cross-env-shell CALYPSO_CLIENT=true NODE_PATH=$NODE_PATH:server:client:. node --max_old_space_size=8192 ./node_modules/.bin/webpack --display-error-details",
```
This reverts @DanReyLop's change from #22623 that makes `build-client` work on Windows. Quote:

> `./node_modules/.bin/webpack` is a Bash script that doesn't work on Windows. The correct Windows script would be `./node_modules/.bin/webpack.cmd`. Instead, I changed it to `./node_modules/webpack/bin/webpack.js`, which will work everywhere.

It's interesting that the `build-server` command uses just `cross-env-shell webpack` and Windows folks don't complain. Maybe that's the way to go? The `cross-env-shell` bit was added by @nylen in #25188.

blowery replied:
build-server is also calling `webpack`, not the bin script directly... I wonder if npm is working some magic in `run` to pick either a .sh or a .cmd based on the host OS...

flootr replied:
Yes, that's what npm is doing for `build-server`. We use `node` to execute webpack in `build-client` and pass on additional arguments for node which is why it doesn't work here. Reverting that part of the script.

flootr committed:
Update package.json

Go back to explicitly call `./node_modules/webpack/bin/webpack.js` in `build-client` to ensure windows compatibility

blowery reviewed (approved):
reapproved :)
